src/java.desktop/macosx/classes/sun/lwawt/macosx/CAccessibleText.java

```diff
@@ -288,7 +288,9 @@ public String call() throws Exception {
                 final AccessibleEditableText aet = ac.getAccessibleEditableText();
                 if (aet == null) return null;
 
-                return aet.getTextRange(location, location + length);
+                int currentLength = aet.getCharCount();
+                return aet.getTextRange(Math.min(currentLength, location),
+                        Math.min(currentLength, location + length));
```

Comment:
It will be good to update the spec of the AccessibleEditableText by this or separate PR, right now it does not mention any exceptions.